miniapps/nurbs/nurbs_curveint.cpp

```diff
@@ -0,0 +1,238 @@
+   }
+   KnotVector *kv = new KnotVector(order, ncp);
+
+   for (int i = 0; i < order+1; i++)
+   {
+      (*kv)[i] = 0.0;
+   }
+   for (int i = order+1; i < ncp; i++)
+   {
+      (*kv)[i] = (i-order)/double(ncp-order);
+   }
+   for (int i = ncp ; i < ncp + order + 1; i++)
+   {
+      (*kv)[i] = 1.0;
+   }
+   return kv;
+}
+
+
+
+int main(int argc, char *argv[])
+{
+   // Parse command-line options.
+   OptionsParser args(argc, argv);
+
+   double l       = 1.0;
+   double a       = 0.1;
+   int ncp        = 9;
+   int order      = 2;
+   bool ifbspline = true;
+
+   args.AddOption(&l, "-l", "--box-side-length",
+                  "Height and width of the box");
+   args.AddOption(&a, "-a", "--sine-ampl",
+                  "Amplitude of the fitted sine function.");
+   args.AddOption(&a, "-a", "--sine-ampl",
+                  "Amplitude of the fitted sine function.");
+   args.AddOption(&ncp, "-n", "--ncp",
+                  "Number of control points used over four box sides.");
+   args.AddOption(&order, "-o", "--order",
+                  "Order of the NURBSPatch");
+   args.AddOption(&ifbspline, "-uw", "--unit-weight", "-nw",
+                  "--non-unit-weight",
+                  "Use a weight of one only for the NURBS-patches (default) or not: B-splines");
+
+   // Parse and print commandline options
+   args.Parse();
+   if (!args.Good())
+   {
+      args.PrintUsage(cout);
+      return 1;
+   }
+   args.PrintOptions(cout);
+
+   if (order < 2 && ifbspline == false)
+   {
+      mfem_error("For a non unity weight, the order should be at least 2.");
+   }
+
+   KnotVector *kv_o1 = UniformKnotVector(1, 2);
+   KnotVector *kv    = UniformKnotVector(order, ncp);
+
+   // 1. Create a box shaped NURBS patch
+   NURBSPatch patch(kv_o1, kv_o1, 3);
+
+   // Set weights
+   for (int j = 0; j < 2; j++)
+      for (int i = 0; i < 2; i++)
+      {
+         patch(i,j,2) = 1.0;
+      }
+
+   // Define patch corners which are box corners
+   patch(0,0,0) = -0.5*l;
+   patch(0,0,1) = -0.5*l;
+
+   patch(1,0,0) = 0.5*l;
+   patch(1,0,1) = -0.5*l;
+
+
+   patch(0,1,0) = -0.5*l;
+   patch(0,1,1) = 0.5*l;
+
+   patch(1,1,0) = 0.5*l;
+   patch(1,1,1) = 0.5*l;
+
+   // 2. Interpolation process
+   Array<Vector*> xy(2);
+   xy[0] = new Vector();
+   xy[1] = new Vector();
+   Vector xi_args, u_args;
+   Array<int> i_args;
+   xy[0]->SetSize(ncp); xy[1]->SetSize(ncp);
+
+   // Refine direction which has fitting
+   patch.DegreeElevate(0, order-kv_o1->GetOrder());
+   if (!ifbspline)
+   {
+      patch(1,0,2) = sqrt(2)/2;
+   }
+   patch.KnotInsert(0, *kv);
+
+   if (ifbspline)
+   {
+      // We locate the controlpoints at the location of the
+      // maxima of the knotvectors. This works very well
+      // for patches with the weight being one.
+      kv->FindMaxima(i_args,xi_args, u_args);
+
+      for (int i = 0; i < ncp; i++)
+      {
+         (*xy[0])[i] = u_args[i]*l;
+         (*xy[1])[i] = a * sin((*xy[0])[i]/l*2*3.1415);
```

Comment:
You can use `M_PI` instead of `3.1415`. There is another instance below.

Reply:
Resolved in [e31bfca](https://github.com/mfem/mfem/pull/3232/commits/e31bfca43bf510dfc9e0cd51398ea06efa9707c6)